packages/mobile/src/screens/chat-screen/ChatScreen.tsx

```diff
@@ -189,10 +198,20 @@ export const ChatScreen = () => {
   const { params } = useRoute<'Chat'>()
   const { chatId } = params
   const url = `/chat/${encodeUrlName(chatId ?? '')}`
-  const [iconOpacity, setIconOpacity] = useState(ICON_BLUR)
-  const [inputMessage, setInputMessage] = useState('')
+  const [iconOpacity, setIconOpacity] = useState<number>(ICON_BLUR)
+  const [inputMessage, setInputMessage] = useState<string>('')
+  const [shouldShowPopup, setShouldShowPopup] = useState<boolean>(false)
+  const [reactionY, setReactionY] = useState<number>(0)
```

Comment:
definitely supporting pulling these into their own hook, as well as converting to refs when possible. also shouldn't have to manually type these calls to useState, TS should be smart enough to infer from the starting value in most cases (unless it's null)